thefuck/rules/ping.py

```diff
@@ -0,0 +1,19 @@
+# -*- encoding: utf-8 -*-
+from six.moves.urllib.parse import urlparse
+from thefuck.utils import for_app
+
+
+@for_app('ping')
+def match(command):
+    # It only fix if the target host is a valid url
+    try:
+        results = urlparse(command.script_parts[-1])
+        is_valid_url = results.hostname is not None
```

Comment:
Would be nice to extract this to a function, have it memoized, check first if `"cannot resolve"` is part of the output. Then, only if that is the case, call the function, check its result (if the URL is valid) and reuse it in `get_new_command`. Can you please do it?